### ISSUE TITLE
Option to not collapse folder on single click

### ISSUE THREAD

**Issue by surdu** [atom/tree-view#411]
It would be nice if the tree-view exposed a setting to allow you to specify if you want to use single or double clicks in order to collapse/expand folders.

I usually use my mouse to select the folder in which to create a folder/file by pressing `a` or `A`, but I have to re-expand the folder afterwards because tree view collapsed it.

**Comment by mathiasvr** [comment]
You could also make so that folders only collapse if they are already selected. This way you could click a folder to select it and click it again to do regular collapse/expand.

**Comment by kankaristo** [comment]
+1

Atom's tree-view behaves very differently to most other tree views (on Windows and Linux at least). Single click usually selects, double-click (or sometimes click selected) usually opens/expands/collapses. On directories, the arrow next to the directory should use single click, but the directory icon and name should use double-click.

I've started *double-right-clicking* in order to select a file or directory in Atom. It works, but it's a weird thing to do.

Keyboard navigation also works in an unfamiliar way. In Atom, pressing the left cursor key collapses the parent directory. Usually when you press left, it moves to the parent directory and pressing left a second time (when the parent directory is selected) collapses it.

**Comment by yoyurec** [comment]
>very differently to most other tree views (on Windows and Linux at least)

+1
single click is annoying :angry:

**Comment by jasonhinkle** [comment]
+1 - the behavior is inconsistent with any treeview that I've seen before OSX, Windows or *nix. If the maintainers prefer the single-click, it would be nice to at least have an preference/setting to change the behavior.

* Single click on the name or folder icon should highlight, but not toggle the folder state
* Single click on the arrow icon should toggle the folder open/closed
* Double-click on the folder or name are not consistent across apps, but for an IDE I think toggling the folder open/closed would be reasonable
* Double-click on the arrow icon is does nothing

If it's a major issue to have different actions for the folder name vs the arrow icon, then I think it would be acceptable to have an preference/setting so that folders are only toggled with a double-click.

Thanks!

**Comment by jasonhinkle** [comment]
Actually I almost have this all working now, if I can get it clean I'll submit a pull request. This is my first Atom package so I'm trying to see if I can put it in as an optional setting.

I can see the difficulty with clicking the arrow, because the click event is on an li element and I'm not sure if it has the info as to whether you clicked on the bullet point vs the text, but I'm going to try to figure it out.

**Comment by yoyurec** [comment]
> almost have this all working
WOW!

> difficulty with clicking the arrow
not so big problem, IMHO

**Comment by jasonhinkle** [comment]
I have this working to my liking, but i haven't figured out how to add an option to the preferences screen. This is actually my first time using coffeescript so I'm fumbling around a bit. If anybody wants to give it a try:

at the terminal:

    apm install tree-view

You'll get a warning that you are overriding a default package, which is fine to ignore.  Once you have that, go into 

    ~/.atom/packages/tree-view/libs/

Here replace the file `tree-view.coffee` with the version from my repo (or simply redo my edits manually if you prefer).  Here's the changes: https://github.com/jasonhinkle/tree-view/commit/8b8947096c9d2cb5913e0ceb50730fadca72ce3f

To me now the tree-view behaves exactly like I would expect from any other app.  I also changed the single-click file behavior which I didn't like either. I probably killed the ability to click for preview though. If there's any interest from the maintainers I can try to clean this up, create a preference setting and submit a pull request.

If you want to undo it, just do this at the command line and you'll be back to the default:

    apm uninstall tree-view

**Comment by yoyurec** [comment]
nice :+1:

**Comment by janmarek** [comment]
Yes please. This is super annoying. Atom's treeview should behave the same like any other treeview.

**Comment by janmarek** [comment]
Btw I feel like this issue has a wrong label. It should be "bug" :)

**Comment by yoyurec** [comment]
any updates?

**Comment by TrevorSayre** [comment]
@jasonhinkle Did you ever submit a PR?

**Comment by jasonhinkle** [comment]
@TrevorSayre I never did hear from any maintainers that they were interested. I thought I might try to write it as a plugin, but I haven't had the time lately :-(

**Comment by synth3tk** [comment]
Another vote for this. Easily the most unintuitive parts of this program for me. Not sure why you would go against traditional convention for tree views, but it's frustrating, to say the least.

Anyway, tried a few plugins, none seem to do anything to prevent single-click expansion/collapse.

**Comment by hszhsh** [comment]
I have written a package to solve this problem: https://atom.io/packages/dbclick-tree-view
You can install it with command:

    apm install dbclick-tree-view

**Comment by dohkoo** [comment]
Thanks hszhsh, this is perfect!

**Comment by vjpr** [comment]
@hszhsh Can you make it only for folders? I like to have single click for file.

**Comment by hszhsh** [comment]
@vjpr, you can turn it off in settings.

**Comment by vjpr** [comment]
@hszhsh Awesome, thanks!

**Comment by hex007** [comment]
@50Wliu Any chance this will be implemented in `tree-view` ?

**Comment by issmith962** [comment]
Love this guy. Just writes a package to fix it and made my day a lot better, 5 years later..
